src/crimson/common/interruptible_future.h

```diff
@@ -578,7 +567,8 @@ class [[nodiscard]] interruptible_future_detail<InterruptCond, seastar::future<T
   [[gnu::always_inline]]
   Result then_wrapped(Func&& func) {
     return core_type::then_wrapped(
-      [func=std::move(func), interrupt_condition=interrupt_cond<InterruptCond>]
+      [func=std::move(func),
+      interrupt_condition=interrupt_cond<InterruptCond>.interrupt_cond]
```

Comment:
Why there is no `ceph_assert(interrupt_cond<InterruptCond>.interrupt_cond);`?

If it's possible here that `interrupt_cond` can be null, then which scenario is that?